sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1865,3 +1865,22 @@ def test_multinomial_identifiability_on_iris(fit_intercept):
     assert_allclose(clf.coef_.sum(axis=0), 0, atol=1e-10)
     if fit_intercept:
         clf.intercept_.sum(axis=0) == pytest.approx(0, abs=1e-15)
+
+
+@pytest.mark.parametrize("multi_class", {'ovr', 'multinomial', 'auto'})
+def test_sample_weight_not_modified(multi_class):
+    X, y = load_iris(return_X_y=True)
+    np.random.seed(1234)
+    W = np.random.random(len(X)) * 10.0
+
+    for weight in [{0: 1.0, 1: 10.0, 2: 1.0}]:
+        for class_weight in (weight, 'balanced'):
+            expected = W.sum()
+
+            clf = LogisticRegression(random_state=0,
+                                    class_weight=class_weight,
+                                    max_iter=200,
+                                    multi_class=multi_class)
+            clf.fit(X, y, sample_weight=W)
+            actual = W.sum()
+            assert expected == actual, 'Sum of weight before ({}) should be the same as sum if weight after ({})'.format(expected, actual)
```

Comment:
We need to keep lines shorter than 80 char, and also to avoid exact float comparison

```suggestion
            msg = (
                f'Sum of weight before ({expected}) should be the same as'
                f'sum if weight after ({actual})'
            )
            assert_allclose(expected, actual, err_msg=msg)
```